Remotion/Data/DomainObjects.UnitTests/Mapping/MappingConfigurationExtensionsTest.cs

```diff
@@ -0,0 +1,170 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Mapping;
+using Remotion.Development.UnitTesting.NUnit;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Mapping
+{
+  [TestFixture]
+  public class MappingConfigurationExtensionsTest
+  {
+    private interface IDummyInterface
+    {
+    }
+
+    private class DummyClass
+    {
+    }
+
+    [Test]
+    public void GetTypeDefinition ()
+    {
+      var typeDefinition = TypeDefinitionObjectMother.CreateClassDefinition();
+      var mappingConfigurationMock = new Mock<IMappingConfiguration>(MockBehavior.Strict);
+      mappingConfigurationMock.Setup(e => e.GetTypeDefinition(typeof(IDummyInterface), It.IsAny<Func<Type, Exception>>())).Returns(typeDefinition);
+
+      var result = mappingConfigurationMock.Object.GetTypeDefinition(typeof(IDummyInterface));
+      Assert.That(result, Is.SameAs(typeDefinition));
+
+      mappingConfigurationMock.VerifyAll();
+    }
+
+    [Test]
+    public void GetTypeDefinition_WithNonExistentType_Throws ()
+    {
+      Assert.That(
+          () => MappingConfiguration.Current.GetTypeDefinition(typeof(IDummyInterface)),
+          Throws.TypeOf<MappingException>()
+              .With.Message.EqualTo("Mapping does not contain type 'Remotion.Data.DomainObjects.UnitTests.Mapping.MappingConfigurationExtensionsTest+IDummyInterface'."));
+    }
+
+    [Test]
+    public void ContainsClassDefinition ()
+    {
+      var mappingConfigurationMock = new Mock<IMappingConfiguration>(MockBehavior.Strict);
+      mappingConfigurationMock.Setup(e => e.ContainsTypeDefinition(typeof(DummyClass))).Returns(true);
```

Comment:
`.VerifyAll()` does not require `.Verifiable()`

Reply:
Oh, that's right. Durign the Migration, we chose Verify() isntead of VerifyAll() because we wanted to separate stub vs mock calls. Do you think we should start to mix those two sematics in the codebase, chosing which ever one is the local optimum? 

If we chose to use Verify instead of VerifyAll, we probably should add a roslyn analyzer to prevent missuse. 

If we chose to mix Verify and VerifyAll, we need to be aware of the semantic difference when changing existing tests.

Thoughts?